torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -4334,6 +4335,46 @@ Error PyTorchModelLoader::loadTranspose(const torch::jit::Node *ptNode) {
   RETURN_ERR(addValueMapping(outputs[0], output->getResult(), dtype));
 }
 
+Error PyTorchModelLoader::loadCopy(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 3, outputs, 1));
+
+  glow::NodeValue lhs, rhs;
+  ASSIGN_VALUE_OR_RETURN_ERR(lhs, getGlowNodeValueForValue(inputs[0]));
+  ASSIGN_VALUE_OR_RETURN_ERR(rhs, getGlowNodeValueForValue(inputs[1]));
+
+  bool nonBlocking = false;
+  ASSIGN_VALUE_OR_RETURN_ERR(
+      nonBlocking,
+      iValToBool(getGlowIValueForValue(ptNode->namedInput("non_blocking"))));
+  RETURN_ERR_IF_NOT(nonBlocking == false,
+                    "non_blocking == True for aten::copy_ not supported.");
+
+  // Handle case where lhs and rhs have different datatypes
+  const bool needsConvertTo =
+      lhs.getType()->getElementType() != rhs.getType()->getElementType();
+  ConvertToNode *toNode = nullptr;
+  if (needsConvertTo) {
+    auto toType =
+        F_.getParent()->uniqueTypeWithNewShape(lhs.getType(), rhs.dims());
+    toNode = F_.createConvertTo("convert_to", rhs, toType);
+  }
+
+  // Handle case where rhs needs to be broadcasted to match lhs
+  const bool needsBroadcast = lhs.getType()->dims() != rhs.getType()->dims();
+  std::vector<NodeValue> broadcastedInputs;
+  if (needsBroadcast) {
+    broadcastedInputs = F_.broadcastInputs(
+        /* axis */ -1, {lhs, needsConvertTo ? toNode->getResult() : rhs});
+  }
+
+  RETURN_ERR(addValueMapping(
```

Comment:
Can we reorganize the logic here a little bit to evade using so much "...?...:..."? Like, having a temporary glow::NodeValue to store the intermediate result.

Reply:
Done.